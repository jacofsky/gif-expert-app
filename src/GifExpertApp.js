import React,{useState} from 'react'

import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'

const GifExpertApp = ({defaultCategories = ['One Punch']}) => {

    

    const [categories, setCategories] = useState(defaultCategories)


    return (
        <>
            <h1 className='title'>GifExpertApp</h1>
        
            <AddCategory setCategories={setCategories}/>

            <hr />

            
            
            
            <ul>
                {categories.map(category => <GifGrid key={category} category={category}/>)}
            </ul>
        </>
    )
}

export default GifExpertApp
