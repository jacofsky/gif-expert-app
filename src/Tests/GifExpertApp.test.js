import react from "react";
import { shallow } from "enzyme";
import GifExpertApp from '../GifExpertApp'

describe('Tests in GifExpertApp.test.js', () => {
    
    
    test('should do the snapshot', () => {    
        
        const wrapper = shallow(<GifExpertApp />)
        
        expect(wrapper).toMatchSnapshot();
    })

    test('should show a list of categories', () => {

        const categories = ['One Punch', 'Dragon Ball']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)    
    })
    
    
})
