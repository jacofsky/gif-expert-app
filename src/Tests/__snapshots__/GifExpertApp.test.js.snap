// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifExpertApp.test.js should do the snapshot 1`] = `
<Fragment>
  <h1
    className="title"
  >
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ul>
</Fragment>
`;

exports[`Tests in GifExpertApp.test.js should show a list of categories 1`] = `
<Fragment>
  <h1
    className="title"
  >
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ul>
</Fragment>
`;
